.github/workflows/upload-and-deploy.yaml

```diff
@@ -48,21 +48,60 @@ jobs:
         uses: aws-actions/configure-aws-credentials@v1
         with:
           aws-region: us-east-1
-          role-to-assume: arn:aws:iam::914833433684:role/sagebase-github-oidc-sage-bion-ProviderRolerecover-VTOKECAKJUBV
+          role-to-assume: ${{ vars.AWS_CREDENTIALS_IAM_ROLE }}
           role-session-name: GitHubActions-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
           role-duration-seconds: 1200
 
       - name: Setup sam
         uses: aws-actions/setup-sam@v2
 
       - name: Set namespace for non-default branch or for tag
-        if: github.ref_name != 'main'
```

Comment:
Hmm, so thinking about this more, I think we should keep this.   Is the rationale of not deploying the main branch to the development AWS account that any feature branches will be tested in the development account?

It would be nice if there was always an exact replica of the infrastructure that is in develop account as it is in prod account. I don't currently see how this would be created:

```
recover-dev-cloudformation/recover/recover...
```

Reply:
I think we want to mirror our other buckets (parquet, intermediate) in that non-main branch deployments will always be namespaced in the dev AWS account because currently in the BridgeDownstream dev account, the cloudformation bucket's contents are not namespaced whilest the other bucket's contents are. I'm not sure of the original rationale but I would support the above of testing feature branches only in dev.

I think the structure should be the same, just that the infrastructure in the prod cloudformation bucket won't be namespaced:

- dev account: recover-dev-cloudformation/etl-416/recover/src/...
- prod account: recover-cloudformation/recover/src/